broadcaster/broadcaster.go

```diff
@@ -76,9 +79,18 @@ func (b *Broadcaster) Broadcast(msg BroadcastMessage) {
 }
 
 func (b *Broadcaster) Confirm(seq arbutil.MessageIndex) {
+	// Always send confirmation for previous sequence number even if current sequence number invalid
+	log.Debug("confirming previous sequence number", "previous", b.previousConfirmedSequenceNumber, "new", seq)
 	b.server.Broadcast(BroadcastMessage{
 		Version:                        1,
-		ConfirmedSequenceNumberMessage: &ConfirmedSequenceNumberMessage{seq}})
+		ConfirmedSequenceNumberMessage: &ConfirmedSequenceNumberMessage{b.previousConfirmedSequenceNumber}})
+
+	if seq <= b.previousConfirmedSequenceNumber {
+		// Already confirmed, don't update previous
+		log.Error("attempt to confirm old sequence number", "previous", b.previousConfirmedSequenceNumber, "new", seq)
```

Comment:
Won't this wait an additional confirmation period for each relay inbetween the sequencer and the end user node? I think we should consider just having the inbox tracker read what the previous batch's message count was.

Reply:
Much better way of handling things, updated